ReleaseProcessScript/tools/Core/git_base_functions.ps1

```diff
@@ -63,9 +63,9 @@ function Push-To-Repos ($Branchname, $WithTags)
   $BeforeBranchname = Get-Current-Branchname
   git checkout $Branchname 2>&1 --quiet
 
-  if ($WithTags)
+  if ( (-not [string]::IsNullOrEmpty($Tagname)) -and (-not (Get-Tag-Exists $Tagname)))
   {
-    $PostFix = "--follow-tags"
+    throw "Tag with name '$($Tagname)' does not exist, abort pushing branch and tag."
```

Comment:
There are multiple options here and I do not know which one is preferable. One could throw if a non-existent tagname is given (just like in the code above) or just "log" that the tag does not exist and only the branch will be pushed. In the case of throwing, one could also argue that it would be nice to checkout `$BeforeBranchname` before throwing (btw, should I rename it to `$PreviousBranchname`?).

There possibly are more even alternatives. I suppose it would be better _not_ to throw and just to push no tag - or what should be the way to go?

Reply:
Actually, I would argue that if somebody made the effort to pass a specific branch name (even though in this case no real person is directly calling this function), if that branch name doesn't exist it's most likely some kind of error worthy of an exception